pkg/controllers/subnet/subnet_controller.go

```diff
@@ -155,7 +155,7 @@ func (r *SubnetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		specChanged = true
 	}
 
-	if subnetCR.Spec.IPv4SubnetSize == 0 {
+	if subnetCR.Spec.IPv4SubnetSize == 0 && len(subnetCR.Spec.IPAddresses) == 0 {
```

Comment:
Why we need to update IPv4SubnetSize even if IPAddresses is not set, can we leave it empty

Reply:
to be compatible with previous behavior.
if only ipAddressBlockVisibility provided in crd, the allocationSize value will be set default size